src/uu/cp/src/platform/linux.rs

```diff
@@ -159,19 +175,53 @@ pub(crate) fn copy_on_write(
             copy_debug.reflink = OffloadReflinkDebug::No;
             sparse_copy(source, dest)
         }
-        (ReflinkMode::Never, _) => {
+        (ReflinkMode::Never, SparseMode::Never) => {
             copy_debug.sparse_detection = SparseDebug::No;
             copy_debug.reflink = OffloadReflinkDebug::No;
             std::fs::copy(source, dest).map(|_| ())
         }
+        (ReflinkMode::Never, SparseMode::Auto) => {
+            copy_debug.sparse_detection = SparseDebug::No;
+            copy_debug.reflink = OffloadReflinkDebug::No;
+            let mut sparse_flag = false;
+            let res = check_sparse_detection(source, &mut sparse_flag);
+            if res.is_ok() {
+                if sparse_flag {
+                    sparse_copy(source, dest)
+                } else {
+                    std::fs::copy(source, dest).map(|_| ())
+                }
+            } else {
+                res
+            }
+        }
```

Comment:
more rust idiomatic:

```suggestion
match check_sparse_detection(source, &mut sparse_flag) {
    Ok(_) if sparse_flag => sparse_copy(source, dest),
    Ok(_) => std::fs::copy(source, dest).map(|_| ()),
    Err(e) => Err(e),
}
```

Reply:
with the adjustement on the return of the value